pkg/reconciler/v1alpha1/eventlistener/eventlistener_test.go

```diff
@@ -246,6 +246,121 @@ func makeDeployment(ops ...func(d *appsv1.Deployment)) *appsv1.Deployment {
 	return &d
 }
 
+// makeDeploymentForTLSConnection is a helper to build a Deployment that is created by an EventListener.
+// It generates a basic Deployment for the simplest EventListener and accepts functions for modification
+func makeDeploymentForTLSConnection(ops ...func(d *appsv1.Deployment)) *appsv1.Deployment {
```

Comment:
Instead of adding this function, is it possible to use makeDeployment() and then pass it a function that contains the modifications for the TLS certs? If we want to call it multiple times, we can extract that function and call it say, withTLSVolumes and then the call can look like `makeDeployment(withTLSVolumes)`. What do you think?

Reply:
Actually we can make use of `makeDeployment()` but we need to pass lot of things to that 
ex: 
* ContainerPort
* Schema, Port to LivenessProbe and ReadinessProbe
* Modify port in the Args
* Modification in VolumeMount, Env and Volume

Basically i need to pass entire `Container` info so that is why created new deployment function.

Shall i go ahead with passing `container` info so that we will have single function `makeDeployment()`

Reply:
right, so what I was suggesting was:

```go

func WithTLSConfig(d *appsv1.Deployment) {
   d.Container.Port = ...
  // All of the TLS specific modifications to the container
}
```

and then we call:

```
makeDeployment(withTLSConfig)
```

Is this what you were thinking off too?

Reply:
Yes 
Modified 
Thank you :+1: